Add a summary table to the access control page (#460)
memfault/memfault-docs: docs/platform/access-control-teams-roles.mdx

This document shows how Organization Admins may manage access control in an

> Organization by using Teams and Roles.

Reviewer: Consider prefixing with a title like `## Quick Overview`

Author: good call, added!

> | Role Type    | Access Level   | Privileges/Notes                            |

Reviewer: I know `Access` doesn't always apply but IMO "notes" makes parsing the rest of the table harder.

Also, fixed nomenclature:

- ~Access Level~ -> Role
- ~Privileges~ -> Access Level

"Access Level" isn't really a Memfault :tm: term but we use the word "access" a lot in this document. 

```suggestion
| Role Type    | Role   | Access Level                            |
```

| ------------ | -------------- | ------------------------------------------- |
| Team         | -              | groups of users:Project Role mappings       |
| Organization | 🙆 **Member**  | basic access to the Organization            |
| Organization | 🛠️ **Admin**   | full access to everything                   |
| Project Role | 🚫 **No Role** | no access to a Project                      |
| Project Role | 📖 **Reader**  | read-only access to a Project               |

> | Project Role | 👷 **Member**  | write access to a Project, _except_ for OTA |

Author: ```suggestion
| Project Role | 📝 **Member**  | write access to a Project, _except_ for OTA |
```

> | Project Role | 🚫 **No Role** | no access to a Project                      |
> | Project Role | 📖 **Reader**  | read-only access to a Project               |
> | Project Role | 👷 **Member**  | write access to a Project, _except_ for OTA |
> | Project Role | 🧑‍🔧 **Manager** | **Member** access + OTA management          |

Reviewer: ```suggestion
| Project Role | 🚫 **No Role** | Team membership grants **no access** to a Project (users may still have access due to membership of a different Team)                     |
| Project Role | 📖 **Reader**  | Team membership grants **read access** to a Project               |
| Project Role | 👷 **Member**  | Team membership grants **write access* to a Project, but not for OTA |
| Project Role | 🧑‍🔧 **Manager** | Team membership grants **write access** to a Project, including OTA          |
```

Author: included, since it overlapped with the other ones did it manually 🫶
